crates/cadenza-eval/src/eval.rs

```diff
@@ -596,14 +605,97 @@ pub fn builtin_assign() -> BuiltinMacro {
                 _ => {
                     // LHS is neither a macro application nor an identifier
                     Err(Diagnostic::syntax(
-                        "left side of = must be an identifier or a macro application (e.g., let x, fn name, measure unit)",
+                        "left side of = must be an identifier, field access (e.g., record.field), or a macro application (e.g., let x, fn name, measure unit)",
                     ))
                 }
             }
         },
     }
 }
 
+/// Handles field assignment of the form: record.field = value
+///
+/// The apply expression represents the field access (e.g., `record.field`),
+/// and rhs_expr is the value to assign.
+///
+/// **Note**: This currently only supports direct variable field assignment
+/// (e.g., `record.field = value`). Chained field assignment (e.g., `obj.a.field = value`)
+/// is not yet supported - the record must be a variable name, not a nested field access.
+fn handle_field_assignment(
+    apply: &Apply,
+    rhs_expr: &Expr,
+    ctx: &mut EvalContext<'_>,
+) -> Result<Value> {
+    // Field assignment requires exactly 2 arguments in the apply: record and field
+    let args = apply.all_arguments();
+    if args.len() != 2 {
+        return Err(Diagnostic::syntax(
+            "field assignment requires exactly record and field name",
+        ));
+    }
+
+    // Get the record identifier (first argument)
+    let record_name = match &args[0] {
+        Expr::Ident(ident) => {
+            let text = ident.syntax().text();
+            let id: InternedString = text.to_string().as_str().into();
+            id
+        }
+        _ => {
+            return Err(Diagnostic::syntax(
+                "field assignment requires a variable name for the record",
+            ));
+        }
+    };
```

Comment:
Why does this need to be an identifier? Why couldn't we accept any arbitrary expression for this?

```
fn make_rec x = { x }
(make_rec 1).x
```